### PR TITLE
Update typeof.md

### PR DESCRIPTION
**Pull request by ghost**
[typeof (C# Reference)](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/typeof)

The original code is not valid because `string s` is assigned a `Type` object (returned by `GetInterface`).
Furthermore, the user has to assume that `method` is an instance of a `MethodInfo` type.

My code provides a real example that will compile without error.

### CODE REVIEW

File: docs/csharp/language-reference/keywords/typeof.md [hunk 1]
```diff
@@ -25,11 +25,22 @@ System.Type type = i.GetType();
   
  The `typeof` operator cannot be overloaded.  
   
- The `typeof` operator can also be used on open generic types. Types with more than one type parameter must have the appropriate number of commas in the specification. The following example shows how to determine whether the return type of a method is a generic <xref:System.Collections.Generic.IEnumerable%601>. Assume that method is an instance of a MethodInfo type:  
+ The `typeof` operator can also be used on open generic types. Types with more than one type parameter must have the appropriate number of commas in the specification. The following example shows how to determine whether the return type of a method is a generic <xref:System.Collections.Generic.IEnumerable%601>. `GetInterface` will return null if the return type is not a IEnumerable<> generic type.
```
Comment (rpetrusha) [inline docs/csharp/language-reference/keywords/typeof.md]: \`GetInterface\` --> \<xref:System.Type.GetInterface%2A?displayProperty=nameWithType> 
null --> \`null\`
is not a --> is not an
The second IEnumerable\<> should either be fenced or an xref (like the first one)

File: docs/csharp/language-reference/keywords/typeof.md [hunk 2]
```diff
@@ -25,11 +25,22 @@ System.Type type = i.GetType();
...
   
 ```csharp  
-string s = method.ReturnType.GetInterface  
-    (typeof(System.Collections.Generic.IEnumerable<>).FullName);  
+// using System;
```
Comment (rpetrusha) [inline docs/csharp/language-reference/keywords/typeof.md]: It strikes me, if the using statements are to be included, that it's best to enclose the entire compilable code example (with a class statement and Main method).

**Review by rpetrusha — CHANGES_REQUESTED** [review]
Thanks for noticing and correcting the error, @johnnyagerard. I've left some suggested changes.

**Comment by johnnyagerard** [comment]
@rpetrusha I tried to add a new code file. I'm not sure I did it the right way.

**Comment by rpetrusha** [comment]
Unfortunately, @johnnyagerard, new code files go in the dotnet/samples repo, so they require a separate PR. The path in the repo would be snippets\csharp\keywords\typeof\. In the PR description, note that the PR is related to dotnet/docs#7269 (the repo name has to be included with the PR number). The reference to the example then takes the form ~/samples/snippets/csharp/keywords/typeof/typeof_3.cs.
 
Let me know if you have questions.

**Comment by johnnyagerard** [comment]
@rpetrusha Thank you for your assistance.

**Comment by rpetrusha** [comment]
Closing and reopening to begin new build after dotnet/samples#265 was merged.

**Review by rpetrusha — APPROVED** [review]
Thank you, @johnnyagerard, for the extra effort you've put into improving this article. I'll merge your PR now.

**Pull request merged**